packages/shared/src/hooks/useWindowEvents.ts

```diff
@@ -1,14 +1,44 @@
-import { useEffect } from 'react';
+import { useCallback, useEffect } from 'react';
+import { AuthEvent } from '../lib/kratos';
 
-// to make intellisense work on the first parameter
-const useWindowEvents: typeof window.addEventListener = (
-  event: string,
-  func: (e: Event) => void,
+interface EventData {
+  eventKey?: AuthEvent | string;
+}
+
+type EventParameter = keyof WindowEventMap;
+type KeyedWindowEventHandler<T = unknown> = (e: MessageEvent<T>) => void;
+
+export interface SocialRegistrationFlow extends EventData {
+  flow: string;
+}
+
+const useWindowEvents = <T extends EventData = EventData>(
+  event: EventParameter,
+  func: KeyedWindowEventHandler<T>,
 ): void => {
   useEffect(() => {
     window.addEventListener(event, func);
     return () => window.removeEventListener(event, func);
   }, [event, func]);
 };
 
+export const useKeyedWindowEvent = <T extends EventData = EventData>(
```

Comment:
Should we not just deprecate the above one in favor of making this one "useWindowEvent" just so we can sole rely on it?

Reply:
Good point. Let me simplify.